File: src/2325.decode-the-message.2406/solution.ts
function decodeMessage (key: string, message: string): string {
    const map = new Map<string, string>();
    for (const c of key) {
        if (c === ' ' || map.has(c)) {
            continue;
        }
        map.set(c, String.fromCharCode(map.size + 97));
    }
    map.set(' ', ' ');

    return message.split('').map((c) => map.get(c)).join('');
}

File: src/2325.decode-the-message.2406/solution.java
class Solution {
    public String decodeMessage(String key, String message) {
        boolean[] set = new boolean[26];
        char[] chars = new char[26];
        int index = 0;

        for(char c : key.toCharArray()){
            if(c == ' ' || set[c-'a']){
                continue;
            }
            set[c-'a'] = true;
            chars[c-'a'] = (char)(index+'a');
            index++;
        }

        char[] result = message.toCharArray();

        for(int i=0;i< result.length;i++){
            if(result[i] == ' '){
                continue;
            }
            result[i] = chars[result[i]-'a'];
        }

        return String.valueOf(result);
    }
}